lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchResourceManager.java

```diff
@@ -0,0 +1,210 @@
+        .setNodeId(id)
+        .setResourceMemory(memorySize)
+        .setVirtualCores(1)
+        .setRuntimeName(RuntimeIdentifier.RUNTIME_NAME)
+        .build());
+  }
+
+  public void onResourceLaunched(ResourceLaunchEvent resourceLaunchEvent) {
+    // Make the configuration file of the evaluator.
+    final File evaluatorConfigurationFile = new File(this.fileNames.getEvaluatorConfigurationPath());
+    try {
+      this.configurationSerializer.toFile(resourceLaunchEvent.getEvaluatorConf(), evaluatorConfigurationFile);
+    } catch (final IOException | BindException e) {
+      throw new RuntimeException("Unable to write configuration.", e);
+    }
+
+    final List<String> command = getLaunchCommand(resourceLaunchEvent);
+
+    try {
+      LaunchBatchTaskWithConf(evaluatorConfigurationFile, command);
+    } catch (IOException ex) {
+      LOG.log(Level.SEVERE, "Error submitting Azure Batch request", ex);
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private List<String> getLaunchCommand(final ResourceLaunchEvent launchRequest) {
+    // TODO: Rebuild this command using JavaLaunchCommandBuilder
+    return Collections.unmodifiableList(Arrays.asList(
+        "/bin/sh -c \"",
+        "unzip local.jar;",
+        "java -Xmx256m -XX:PermSize=128m -XX:MaxPermSize=128m -classpath local/*:global/*",
+        "-Dproc_reef org.apache.reef.runtime.common.REEFLauncher reef/local/evaluator.conf",
+        "\""
+    ));
+  }
+
+  private void LaunchBatchTaskWithConf(File evaluatorConf, List<String> command) throws IOException {
+    // TODO: Reuse Job Id to submit the task and avoid using JobManagerTask
+    LOG.log(Level.INFO, "onResourceLaunched LaunchBatchTaskWithConf ", StringUtils.join(command, ' '));
+    BatchSharedKeyCredentials cred = new BatchSharedKeyCredentials(
+        this.azureBatchAccountUri, this.azureBatchAccountName, this.azureBatchAccountKey);
+    BatchClient client = BatchClient.open(cred);
+
+    PoolInformation poolInfo = new PoolInformation();
+    poolInfo.withPoolId(this.azureBatchPoolId);
+
+
+    final LocalResource uploadedConfFile = this.azureUploader.uploadFile(evaluatorConf);
+    final ResourceFile confSourceFile = new ResourceFile()
+        .withBlobSource(uploadedConfFile.getUrl())
+        .withFilePath(evaluatorConf.getPath());
+
+    final File localJar = new File("local.jar");
+    final LocalResource jarFile = this.azureUploader.uploadFile(localJar);
+    final ResourceFile jarSourceFile = new ResourceFile()
+        .withBlobSource(jarFile.getUrl())
+        .withFilePath(localJar.getPath());
+
+    List<ResourceFile> resources = new ArrayList<>();
+    resources.add(confSourceFile);
+    resources.add(jarSourceFile);
+
+    JobManagerTask jobManagerTask = new JobManagerTask()
+        .withId(this.applicationId)
+        .withResourceFiles(resources)
+        .withCommandLine(StringUtils.join(command, ' '));
+
+    JobAddParameter jobAddParameter = new JobAddParameter()
```

Comment:
I suggest we use the JOB ID environment variable instead of creating a new job.

Reply:
mentioned in Note. This will be implemented in follow up task.

Reply:
If nothing else, I'd suggest this to be one thing you fix before checking in. Getting the job ID from the environment variable will be one line change and adding a job to a task will be easier than creating a new job. Plus, it will unblock me for the rest of the day :)
(Otherwise, I'll have to do once I pull your change)

Reply:
Sure. Not a problem.

Reply:
Thank you!